skt/kernelbuilder.py

```diff
@@ -234,11 +249,73 @@ def assemble_make_options(self):
         """Assemble all of the make options into a list."""
         kernel_build_argv = (
             self.make_argv_base
-            + self.targz_pkg_argv
+            + self.compile_args
             + self.extra_make_args
         )
         return kernel_build_argv
 
+    def find_rpm(self):
+        """
+        Find RPMs in the buildlog.
+
+        Returns:
+            The full path to the RPM repository.
+
+        """
+        fpath = None
+
+        # Compile a regex to look for the tarball filename.
+        rpm_regex = re.compile(r"^Wrote: (.*\.rpm)$")
+
+        # Read the buildlog line by line looking for the RPMs.
+        rpm_files = []
+        with open(self.buildlog, 'r') as fileh:
+            for log_line in fileh:
+                match = rpm_regex.search(log_line)
+
+                # If we find a match, store the path to the tarball and stop
+                # reading the buildlog.
+                if match:
+                    rpm_path = match.group(1)
+                    rpm_files.append(rpm_path)
+
+                    # Ensure the RPM is actually on the filesystem.
+                    if not os.path.isfile(rpm_path):
+                        raise IOError(
+                            "Built kernel RPM not found: "
+                            "{}".format(rpm_path)
+                        )
+
+        if rpm_files:
+            fpath = self.make_rpm_repo(rpm_files)
+
+        return fpath
+
+    def make_rpm_repo(self, rpm_files):
+        """
+        Make a RPM repository within a directory.
+
+        Args:
+            rpm_files:  A list of RPM paths.
+
+        Returns:
+            Path to the repo directory.
+
+        """
+        # Create a directory for the RPM repo.
+        repo_dir = join_with_slash(self.source_dir, 'rpm_repo/')
+        os.mkdir(repo_dir)
```

Comment:
repo_dir must not exist every time you're trying to build rpms using skt. Not saying that it's wrong but is that the behavior we want/expect?

Reply:
I could put a check in there to see if it exists and skip that step if so.